### ISSUE TITLE
undefined rotate() called in matrix_char.cpp

### ISSUE THREAD

**Issue by skuzzymiglet** [txstc55/matrix_viewer#2]
https://github.com/txstc55/matrix_viewer/blob/master/src/matrix_char.cpp calls `rotate()` on line 98. `make` fails with

```
matrix_viewer/src/matrix_char.cpp:98:5: error: ‘rotate’ was not declared in this scope
   98 |     rotate(color_map.begin(), color_map.begin() + max_y - 1, color_map.end());
      |     ^~~~~~

```

**Comment by txstc55** [comment]
This is probably related to not including algorithm. Can you try to include algorithm and see if that fixed the issue

**Comment by skuzzymiglet** [comment]
What is algorithm? Is it a package I can install? Do I need to modify the code?

**Comment by txstc55** [comment]
just do 
```c++
#include <algorithm>
```
at the beginning of matrix_char.cpp, what system are you trying to build with and are you using clang or gcc?

**Comment by skuzzymiglet** [comment]
```
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/lto-wrapper
Target: x86_64-pc-linux-gnu
Configured with: /build/gcc/src/gcc/configure --prefix=/usr --libdir=/usr/lib --libexecdir=/usr/lib --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=https://bugs.archlinux.org/ --enable-languages=c,c++,ada,fortran,go,lto,objc,obj-c++,d --enable-shared --enable-threads=posix --with-system-zlib --with-isl --enable-__cxa_atexit --disable-libunwind-exceptions --enable-clocale=gnu --disable-libstdcxx-pch --disable-libssp --enable-gnu-unique-object --enable-linker-build-id --enable-lto --enable-plugin --enable-install-libiberty --with-linker-hash-style=gnu --enable-gnu-indirect-function --enable-multilib --disable-werror --enable-checking=release --enable-default-pie --enable-default-ssp --enable-cet=auto gdc_include_dir=/usr/include/dlang/gdc
Thread model: posix
gcc version 9.2.0 (GCC)
```

**Comment by txstc55** [comment]
fixed at #3
